### ISSUE TITLE
S4017: Don't count Nullable<T> shorthand when used as a generic type parameter

### ISSUE THREAD

**Issue by saciervo** [SonarSource/sonar-dotnet#3277]
### Description

Rule S4017 is violated if we nest a nullable type in a generic type. The rule description states that calling a method with such a nested type argument requires complicated and confusing code. Imho, this is not the case with `Nullable<T>` and even more if the shorthand is used. E.g `decimal?` instead of `Nullable<decimal>`. 

Therefore, I would like to suggest that `Nullable<T>` is allowed as a nested generic type or maybe even better; Allow `Nullable<T>`, when the shorthand is used.

### Repro steps

```csharp
public static PropertyBuilder<decimal?> HasPrecision(this PropertyBuilder<decimal?> builder, int precision, int scale)
{
    if (builder == null)
    {
        throw new System.ArgumentNullException(nameof(builder));
    }

    return builder.HasColumnType($"decimal({precision},{scale})");
}
```

### Related information

* SonarAnalyzer.CSharp 8.4.0.15306

**Comment by pavel-mikula-sonarsource** [comment]
Hi @saciervo ,

Thank you for reporting this case. I confirm this as False Positive, the rule should not raise when the `decimal?` nullable syntax is used.
